### PR TITLE
Version1

### PR DESCRIPTION
**Pull request by BenoitCier**
## Problem
- *Upgrade of the application to v1.0.1*

## PR Status
- [x] Code finished.
- [ ] Tested with Package_check.
- [] Fix or enhancement tested.
- [x] Upgrade from last version, install and remove tested.
- [x] Can be reviewed and tested.

### CODE REVIEW

**Comment by rodinux** [comment]
Ok, is a good idea to update the version, unfortunately there is a trouble for me trying your code. on a local machine, I have try update garradin with success, but when I try connecting to admin page, I am redirected to SSO page. It seems always a problem with nginx.conf... or perhaps the php.fpm.conf...  So first, we need resolve this mismatch...

**Comment by bohwaz** [comment]
Unfortunately this won't work, old release are not left on the server as it's causing people to email me because they installed an old version. Please don't link to a fixed release, use the API to find out the latest version available: https://github.com/YunoHost-Apps/garradin_ynh/issues/31

**Comment by bohwaz** [comment]
And I just released the 1.0.2, and I'm gonna release a new version every week or every two weeks, so it's not sustainable to have to update your package every time :+1: 

You could do something that would only retain 1.0.* versions for example if you're worried about having a regression or something.

**Comment by rodinux** [comment]
I have similar problem https://github.com/YunoHost-Apps/garradin_ynh/pull/32#issuecomment-755444607 with last version. First we need to understand what's wrong with nginx.conf...

**Comment by BenoitCier** [comment]
I have reproduce the problem, it's only with subpath, i'm readind nginx documentation about rwrite rules...

**Comment by BenoitCier** [comment]
@bohwaz 
I'm not sure the problem was on nginx.conf/yunohost.  
Version 0.9.8    <form method="post" action="https://yunohost.local/test3/admin/install.php">
Version 1.0.2       <form method="post" action="/admin/install.php">
Every other URL on the html file contains the subpath. 

I have try to use the config.local.php to define WWW_URI whitout sucess.
It's a possible to be a  bug into garradin?

**Comment by bohwaz** [comment]
It works for me locally with a manual nginx setup, as I reported here: https://github.com/YunoHost-Apps/garradin_ynh/issues/24#issuecomment-699636299

So probably an issue between nginx and YunoHost, but correct me if I'm wrong.

**Comment by bohwaz** [comment]
> Version 0.9.8 form method="post" action="https://yunohost.local/test3/admin/install.php"
> Version 1.0.2 form method="post" action="/admin/install.php"

Yes the URLs now don't include the host anymore. In `config.local.php` you can customize the root URI:

```
<?php

namespace Garradin;

const WWW_URI = '/test/garradin/www/';
```

You can find the working nginx setup documented at the end of this page: https://fossil.kd2.org/garradin/wiki?name=Installation+sous+Debian-Ubuntu

Multiple people have reported this setup working with nginx so far, with no need to change the config file (unless you're not using a dedicated vhost).

**Comment by rodinux** [comment]
Well, there is yet a script in the installation script to edit the config.local.php adding the ```const WWW_URI```. So it should be rather an redirection error...

nginx.conf looks like this:

```
#sub_path_only rewrite ^__PATH__$ __PATH__/ permanent;
location __PATH__/ {

  alias __FINALPATH__/www/ ;

  if ($scheme = http) {
    rewrite ^ https://$server_name$request_uri? permanent;
  }

  index index.php /_route.php;
  try_files $uri $uri/ __PATH__/__PATH__/_route.php?$query_string;


  location ~ \.php$ {
        if (!-e $request_filename) {
            rewrite ^__PATH__/?(.*)$ __PATH__/_route.php?/$1 last;
            break;
        }
        fastcgi_pass unix:/var/run/php/php__PHPVERSION__-fpm-__NAME__.sock;
        fastcgi_index index.php;
        include fastcgi_params;
        fastcgi_param REMOTE_USER $remote_user;
        fastcgi_param PATH_INFO $fastcgi_path_info;
        fastcgi_param SCRIPT_FILENAME $request_filename;
  }

  # Increase size limit
  client_max_body_size 2M;

  # Include SSOWAT user panel.
  include conf.d/yunohost_panel.conf.inc;
}h 

```
On local machine, I have this log on /var/log/nginx/error.log

```Using 32768KiB of shared memory for nchan in /etc/nginx/nginx.conf:63``` this line is just the end for includes VirtualHost, so I think the problem is with the nginx garradin.conf...

**Comment by BenoitCier** [comment]
I ask @bohwaz  by mail few minute ago about something in the code of garradin. It's solve the problem for me. I wait for his answer.

**Comment by BenoitCier** [comment]
@bohwaz answer me so I will be solve on 1.0.3 (next week?) it's not a yunohost problem. Just work whitout subpath for the moment.

**Comment by BenoitCier** [comment]
> Well, there is yet a script in the installation script to edit the config.local.php adding the `const WWW_URI`.

 I think we can remove this script, I think garradin discover the URI automatically, I test

**Comment by rodinux** [comment]
> @bohwaz answer me so. It will be solve on 1.0.3 (next week?) it's not a yunohost problem.
> Just work whitout subpath for the moment.
> 
> Edit: https://github.com/BenoitCier/garradin_ynh/tree/version102/ : issue solved with Utils.php file (source app)

Yes, this is a solution, I have test with success, thanks.

**Comment by rodinux** [comment]
> > Well, there is yet a script in the installation script to edit the config.local.php adding the `const WWW_URI`.
> 
> I think we can remove this script, I think garradin discover the URI automatically, I test

I need try if it match... it was also for the script change_url...

File: manifest.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
         "en": "Software to manage association",
         "fr": "Logiciel libre de gestion d'association"
     },
-    "version": "0.9.8.1~ynh4",
+    "version": "1.0.2~ynh4",
```
Comment (ericgaspar) [inline manifest.json:9]:
```suggestion
    "version": "1.0.2~ynh1",
```

**Review by ericgaspar — CHANGES_REQUESTED** [review]

File: manifest.json (hunk 1 above)
Comment (rodinux) [reply manifest.json:9]: Ok, you're right.

**Pull request merged**